### PR TITLE
Feature: broadcast vote power

### PR DESCRIPTION
**Pull request by Frozen**
Feature: broadcast vote power

### CODE REVIEW

**Review by ONECasey — APPROVED** [review]

File: consensus/consensus_service.go [hunk 1]
```diff
@@ -153,13 +156,56 @@ func (consensus *Consensus) updateBitmaps() {
...
+			return
+		}
+		msgToSend := proto.ConstructConsensusMessage(marshaledMessage)
+		if err := consensus.msgSender.SendWithoutRetry(
+			[]nodeconfig.GroupID{
+				nodeconfig.NewGroupIDByShardID(nodeconfig.ShardID(consensus.ShardID))},
+			p2p.ConstructMessage(msgToSend),
+		); err != nil {
+			consensus.getLogger().Err(err).
+				Msg("[LastSignPower] could not send out the ViewChange message")
```
Comment (sophoah) [inline consensus/consensus_service.go:195]: please update the log message here

File: node/node.go [hunk 2]
```diff
@@ -619,7 +626,14 @@ func validateShardBoundMessage(consensus *consensus.Consensus, peer libp2p_peer.
...
 			return nil, nil, true, errors.WithStack(errViewIDTooOld)
 		}
-	} else {
+	case maybeSP != nil:
+		if maybeSP.ShardId != consensus.ShardID {
+			nodeConsensusMessageCounterVec.With(prometheus.Labels{"type": "invalid_shard"}).Inc()
+			return nil, nil, true, errors.WithStack(errWrongShardID)
+		}
+		senderKey = maybeSP.SenderPubkey
+		consensus.SetLastKnownSignPower(maybeSP.Commit)
```
Comment (sophoah) [inline node/node.go:635]: we need to capture all 3 VP :
Prepare:      getOrZero(prep),
Commit:       getOrZero(comm),
Change:       getOrZero(view),

during an outage, most likely we won't even hit the commit phase, and during that time only view change VP will be seen.

**Comment by sophoah** [comment]
@Frozen were you able to test that PR in localnet and confirm the VP and api calls are working fine ? hmy_getLastSigningPower at the current state is only working when this is call on the leader, now this one should work on all nodes.

**Comment by Frozen** [comment]
> @Frozen were you able to test that PR in localnet and confirm the VP and api calls are working fine ? hmy_getLastSigningPower at the current state is only working when this is call on the leader, now this one should work on all nodes.

i have hmy_getLastSigningPower and hmy_getLastSigningPower2 methods to compare results, and i'm waiting for watchdog localnet integration to confirm that everything works as expected.

**Comment by mur-me** [comment]
@Frozen, please confirm that you are unblocked and have watchdog locally

**Comment by GheisMohammadi** [comment]
Looks good to me. We need to test the LastSignPowerRequest in the localnet. Other than that, everything is good. Thanks!

**Review by GheisMohammadi — APPROVED** [review]

**Pull request merged**